### ISSUE TITLE
[Bug]: updates to pnpm-lock.yaml are not detected

### ISSUE THREAD

**Issue by leungster** [aspect-build/rules_js#2070]
### What happened?

We're manually updating the `pnpm-lock.yaml` file using `bazel run -- @pnpm//:pnpm --dir $PWD install --lockfile-only` and setting `update_pnpm_lock = False`.

In this setup, changes to the `pnpm-lock.yaml` file do not trigger a re-fetch of node modules. This results in missing link errors and requires running `bazel clean --expunge` to resync node_modules locally.

### Version

Development (host) and target OS/architectures:
Mac M1

Output of `bazel --version`:

bazel 8.0.0

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:

```
bazel_dep(
    name = "aspect_rules_js",
    version = "2.1.2",
)
```

Language(s) and/or frameworks involved:

### How to reproduce

```shell
Setup `npm_translate_lock` as such


npm.npm_translate_lock(
    name = "npm",
    bins = {
        "typescript": [
            "tsc=./bin/tsc",
            "tsserver=./bin/tsserver",
        ],
    },
    generate_bzl_library_targets = True,
    npm_package_target_name = "{dirname}",
    npmrc = "//:.npmrc",
    pnpm_lock = "//:pnpm-lock.yaml",
    update_pnpm_lock = False,
    verify_node_modules_ignored = "//:.bazelignore",
)


1. Run `bazel query` on a `node_modules` dependency in an artifact. Make note of the version for that 
dependency.
2. Bump the version of the dependency and call `bazel run -- @pnpm//:pnpm --dir $PWD install --lockfile-only` to update the lock file
3. Run `bazel query` and see if the dependency version matches package.json.
```

### Any other information?

I believe it's due to `--incompatible_no_implicit_watch_label` being `true` in Bazel 8.
The pnpm lock file attribute isn't watched for changes.

**Comment by jbedard** [comment]
This seems to still be happening in some scenarios so reopening to investigate.

**Comment by jbedard** [comment]
Fixed with https://github.com/aspect-build/rules_js/commit/0b21e0b8c87869ac0fe9e59290357f8b27866f33
